packages/types/src/type/Event.ts

```diff
@@ -119,23 +128,48 @@ export default class Event extends Struct {
 
   // This is called/injected by the API on init, allowing a snapshot of
   // the available system events to be used in lookups
-  static injectMetadata (metadata: MetadataV0): void {
-    metadata.events.forEach((section, sectionIndex) => {
-      const sectionName = stringCamelCase(section.name.toString());
-
-      section.events.forEach((meta, methodIndex) => {
-        const methodName = meta.name.toString();
-        const eventIndex = new Uint8Array([sectionIndex, methodIndex]);
-        const typeDef = meta.arguments.map((arg) => getTypeDef(arg));
-        const Types = typeDef.map(getTypeClass);
-
-        EventTypes[eventIndex.toString()] = class extends EventData {
-          constructor (value: Uint8Array) {
-            super(Types, value, typeDef, meta, sectionName, methodName);
-          }
-        };
+  static injectMetadata (metadata: Metadata): void {
```

Comment:
so this is useless, could have kept the v0 version? I'll overwrite it with MetadataV4 anyways

Reply:
I had conflicts since I changed it from the non-working version, so kept it. (Although I know it will be replaced)